[docs] Update `eas/checkout` docs by informing about the `ref` input (#48233)
expo/expo: docs/pages/eas/workflows/syntax.mdx

        # @end #
```

By default, the step checks out the ref that triggered the job. Use `ref` to check out a different branch, tag, or commit:

```yaml
jobs:
  my_job:
    steps:
      # @info #
      - uses: eas/checkout
        with:
          ref: feature/add-icon
        # @end #
```

> Branches accept a bare name or a qualified ref such as `refs/heads/feature/add-icon`. Tags need the qualified form, such as `refs/tags/v1.2.3`. A full commit SHA also works and leaves the repository on a detached `HEAD`. The step performs a shallow, depth-one fetch from the source repository's `origin`, so the ref must be reachable there.

Reviewer: ⚠️ **correctness** `review-ai-text.md#t7-overclaiming`

A tag also ends on a detached `HEAD`. Only a branch ref checks out onto a named branch, so "A full commit SHA also works and leaves the repository on a detached `HEAD`" reads as if detached `HEAD` were specific to SHAs. Splitting the three ref shapes into a list makes the per-shape behavior land.

```suggestion
`ref` accepts:

- A branch, as a bare name such as `feature/add-icon` or a qualified ref such as `refs/heads/feature/add-icon`. The repository ends up on that branch.
- A tag, as a qualified ref such as `refs/tags/v1.2.3`. The repository ends up on a detached `HEAD`.
- A full commit SHA. The repository ends up on a detached `HEAD`.

The step runs a shallow fetch (`--depth 1`) from the source repository's `origin`, so the ref must be reachable there.
```

<!-- docs-review -->

Author: Addressed in https://github.com/expo/expo/pull/48233/commits/ce3714f51540686b3551e9aaa91c85741e6e3f24

> `ref` only works when your project sources come from a Git repository, for example a job triggered through the GitHub integration. Local builds and uploaded project tarballs do not support it. Place the step before any other step that checks out the project. Setting `ref` after checkout fails the step.

Reviewer: ✏️ **style** `prose.md#core-principles`

In a workflow job, `eas/checkout` is the only step that checks out the project, so "Place the step before any other step that checks out the project" points at nothing a reader can act on. The rule worth stating is that `ref` has to go on the job's first `eas/checkout`.

```suggestion
`ref` only works when your project sources come from a Git repository, for example, a job triggered through the GitHub integration. Local builds and uploaded project tarballs do not support it. Set `ref` on the job's first `eas/checkout` step. Setting it on a later step fails, because the project is already checked out.
```

<!-- docs-review -->
